log/audit-logger.go

```diff
@@ -54,6 +56,37 @@ func NewAuditLogger(log *syslog.Writer, stats statsd.Statter) (*AuditLogger, err
 	return &AuditLogger{log, stats}, nil
 }
 
+// initializeAuditLogger should only be used in unit tests. Failures in this
+// method are unlikely as the defaults are safe, and they are also
+// of minimal consequence during unit testing -- logs get pritned to stdout
+// even if syslog is missing.
+func initializeAuditLogger() {
+	stats, _ := statsd.NewNoopClient(nil)
+	audit, _ := Dial("", "", "default", stats)
+	audit.Notice("Using default logging configuration.")
+
+	SetAuditLogger(audit)
+}
+
+// SetAuditLogger configures the singleton audit logger. This method
+// must only be called once, and before calling GetAuditLogger the
+// first time.
+func SetAuditLogger(logger *AuditLogger) {
+	_Singleton.once.Do(func() {
+		_Singleton.log = logger
+	})
+}
+
+// GetAuditLogger obtains the singleton audit logger. If SetAuditLogger
+// has not been called first, this method initializes with basic defaults.
+// The basic defaults cannot error, and subequent access to an already-set
```

Comment:
It's not clear to me that the basic defaults cannot error. We just happen to ignore their errors in initializeAuditLogger.

Also, initializeAuditLogger says it should only be used in unit tests, but calling it from GetAuditLogger means there's a decent chance it will get called from live code, in the case of a mistake. And we wouldn't necessarily notice.